contracts/test/integration/IntegrationTestVip13.t.sol

```diff
@@ -0,0 +1,197 @@
+import {IPCVGuardian} from "../../pcv/IPCVGuardian.sol";
+
+contract IntegrationTestVIP13 is TimelockSimulation, vip13 {
+    uint256 public constant mintAmountDai = 1_000_000e18;
+    uint256 public constant mintAmountUsdc = 1_000_000e6;
+    uint256 public constant voltMintAmount = 1_000_000e18;
+
+    IERC20 public dai = IERC20(MainnetAddresses.DAI);
+    IERC20 public usdc = IERC20(MainnetAddresses.USDC);
+    ICore private core = ICore(MainnetAddresses.CORE);
+    IVolt oldVolt = IVolt(MainnetAddresses.VOLT);
+
+    function setUp() public {
+        mainnetSetup();
+        simulate(
+            getMainnetProposal(),
+            TimelockController(payable(MainnetAddresses.TIMELOCK_CONTROLLER)),
+            IPCVGuardian(MainnetAddresses.PCV_GUARDIAN),
+            MainnetAddresses.GOVERNOR,
+            MainnetAddresses.EOA_1,
+            vm,
+            false
+        );
+        mainnetValidate();
+
+        vm.startPrank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(this), mintAmountDai);
+        dai.transfer(address(voltV2DaiPriceBoundPSM), mintAmountDai * 2);
+        vm.stopPrank();
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        voltV2.mint(address(this), mintAmountDai);
+        core.revokeMinter(MainnetAddresses.GOVERNOR);
+        vm.stopPrank();
+
+        uint256 balance = usdc.balanceOf(MainnetAddresses.KRAKEN_USDC_WHALE);
+        vm.prank(MainnetAddresses.KRAKEN_USDC_WHALE);
+        usdc.transfer(address(this), balance);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        voltV2.mint(address(voltV2UsdcPriceBoundPSM), voltMintAmount);
+        voltV2.mint(address(this), voltMintAmount);
+        vm.stopPrank();
+
+        usdc.transfer(address(voltV2UsdcPriceBoundPSM), balance / 2);
+    }
+
+    function testSwapDaiForVolt() public {
+        uint256 userStartingVoltBalance = voltV2.balanceOf(address(this));
+        uint256 minAmountOut = voltV2DaiPriceBoundPSM.getMintAmountOut(
+            mintAmountDai
+        );
+        uint256 startingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        dai.approve(address(voltV2DaiPriceBoundPSM), mintAmountDai);
+        uint256 amountVoltOut = voltV2DaiPriceBoundPSM.mint(
+            address(this),
+            mintAmountDai,
+            minAmountOut
+        );
+
+        uint256 endingUserVOLTBalance = voltV2.balanceOf(address(this));
+        uint256 endingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        assertEq(
+            endingUserVOLTBalance,
+            amountVoltOut + userStartingVoltBalance
+        );
+        assertEq(
+            endingPSMUnderlyingBalance - startingPSMUnderlyingBalance,
+            mintAmountDai
+        );
+    }
+
+    function testSwapVoltForDai() public {
```

Comment:
would be great to add these tests to the timelock simulation framework

Reply:
wouldn't that be equivalent to the mint and redeem verification we do in the timelock simulation after the proposal?

Reply:
yes, it's mainly an additional assertion of getting the current peg price and doing verifications that the amountsOut match up with the amount in and oracle price.